### PR TITLE
2020 book display

### PR DESCRIPTION
**Pull request by Raemon**
Note: this adds a `hideFrontpageBook2020Ad` field.

image

image

image

image

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1205281083165806) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by s-cheng** [comment]
So pretty! 😍

File: packages/lesswrong/components/books/Book2020Animation.tsx [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+}) => {
+  const { query } = useLocation();
+  const success = !!query.success
+  return (
+    <div className={classNames(classes.root, {[classes.success]: success})}>
+      <div className={classes.parent}>
+        <img className={classNames(classes.book, classes.one)} src="https://res.cloudinary.com/lesswrong-2-0/image/upload/v1691622592/coordination-constraint_w5obih.png" />
+        <img className={classNames(classes.book, classes.two)}src="https://res.cloudinary.com/lesswrong-2-0/image/upload/v1691622593/alignment-agency-cover_nvzux7.png" />
+        <img className={classNames(classes.book, classes.three)} src="https://res.cloudinary.com/lesswrong-2-0/image/upload/v1691622595/timelines-takeoff-cover_wnb0nc.png" />
+        <img className={classNames(classes.book, classes.four)}src="https://res.cloudinary.com/lesswrong-2-0/image/upload/v1691623651/reality-reason_llvcqx.png" />
```
Comment (b0b3rt) [inline packages/lesswrong/components/books/Book2020Animation.tsx:163]: nit: pretty hard to read without any space between the `className` and `src` attributes

File: packages/lesswrong/components/books/Book2020Animation.tsx [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+      </div>
+      <div className={classes.revealedContent}>
+        { children}
```
Comment (b0b3rt) [inline packages/lesswrong/components/books/Book2020Animation.tsx:166]: nit: asymmetrical spacing

File: packages/lesswrong/components/books/Book2020FrontpageWidget.tsx [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+  },
+  bookExplanation: {
+    float: 'right',
+    textAlign: "right",
+    [theme.breakpoints.down('md')]: {
+      paddingRight: 0
+    },
+    [theme.breakpoints.down('xs')]: {
+      // paddingRight: 16,
+      // textAlign: 'right',
```
Comment (b0b3rt) [inline packages/lesswrong/components/books/Book2020FrontpageWidget.tsx:48]: cleanup?

File: packages/lesswrong/components/books/Book2020FrontpageWidget.tsx [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+          hideFrontpageBook2020Ad: true
+        },
+      })
+    } else {
+      openDialog({
+        componentName: "LoginPopup",
+        componentProps: {}
+      });
+    }
+  }
```
Comment (b0b3rt) [inline packages/lesswrong/components/books/Book2020FrontpageWidget.tsx:116]: Feels like another use-case motivating "combined user setting + cookie infra"

File: packages/lesswrong/components/books/Book2020FrontpageWidget.tsx [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+
+  const BookMarketingText = ({title, subtitle, description, buttons}: {
```
Comment (b0b3rt) [inline packages/lesswrong/components/books/Book2020FrontpageWidget.tsx:118]: Feels kinda weird for this to be defined inside another component, maybe just pull it out?  Can keep it in the same file

File: packages/lesswrong/components/recommendations/LWRecommendations.tsx [hunk 6]
```diff
@@ -167,6 +171,7 @@ const LWRecommendations = ({
     const renderContinueReading = currentUser && (continueReading?.length > 0) && !settings.hideContinueReading
 
     return <SingleColumnSection className={classes.section}>
+      <Book2020FrontpageWidget/>
```
Comment (b0b3rt) [inline packages/lesswrong/components/recommendations/LWRecommendations.tsx:174]: Does this get rendered regardless of `bookDisplaySetting`?

File: packages/lesswrong/lib/collections/users/schema.ts [hunk 7]
```diff
@@ -1722,6 +1722,17 @@ const schema: SchemaType<DbUser> = {
     label: "Hide the frontpage book ad"
   },
 
+  hideFrontpageBook2020Ad: {
+    type: Boolean,
+    canRead: [userOwns, 'sunshineRegiment', 'admins'],
+    canCreate: ['members'],
+    canUpdate: [userOwns, 'sunshineRegiment', 'admins'],
+    optional: true,
```
Comment (b0b3rt) [inline packages/lesswrong/lib/collections/users/schema.ts:1730]: should also be `nullable: true`, if we're not giving it a `schemaDefaultValue`.  (giving it a `schemaDefaultValue` would make the schema migration which creates the column in the database also assign a value for all existing users, so there'd be no need to make it nullable then.)

File: packages/lesswrong/lib/collections/users/fragments.ts [hunk 8]
```diff
@@ -475,7 +476,7 @@ registerFragment(`
     hideFrontpageMap
     hideTaggingProgressBar
     hideFrontpageBookAd
-    hideFrontpageBook2019Ad
+    hideFrontpageBook2020Ad
```
[the remark this answers is not in the document]
Comment (Raemon) [reply packages/lesswrong/lib/collections/users/fragments.ts:479]: In this case it's from 2 years ago

[the remark this answers is not in the document]
Comment (b0b3rt) [reply packages/lesswrong/lib/collections/users/fragments.ts:479]: The errors are caused by a mismatch between the fragment definition that clients have and what they get back from the server, so it'll happen regardless

**Review by b0b3rt — APPROVED** [review]

**Pull request merged**